src/js/providers/utils/play-promise.ts

```diff
@@ -1,40 +1,44 @@
-export default function createPlayPromise(video) {
-    return new Promise(function(resolve, reject) {
+type PlayPromiseError = Omit<DOMException, 'name' | 'code'> & {
```

Comment:
So I wound up remaking the `PlayPromiseError` type because I think it is important to note that it is a DOMException, but since those properties are readonly, they needed some way to be set.
Open to thoughts.